### PR TITLE
Feature: Use ethers signer

### PR DESCRIPTION
**Pull request by mmv08**
This PR:
- Add prettier configuration, dependency updates, format code
- Some linter warning fixes
- Use new safe-apps-sdk-ethers-provider with the signing capabilities

### CODE REVIEW

File: src/components/rebirth.tsx [hunk 1]
```diff
@@ -1,196 +1,229 @@
...
+        );
+        await pushDetails(hashImage, announcementDetails);
+        await module.announceTransaction(
+          announcementDetails.executor,
+          announcementDetails.to,
+          announcementDetails.value,
+          announcementDetails.data,
+          announcementDetails.operation,
+          announcementDetails.nonce,
+        );
```
Comment (mmv08) [inline src/components/rebirth.tsx:163]: actual change

File: src/components/rebirth.tsx [hunk 2]
```diff
@@ -1,196 +1,229 @@
...
+        console.warn({ txHash });
+        const announcementDetails: AnnouncementDetails = await pullDetails(txHash);
+        await module.executeTransaction(
+          announcementDetails.executor,
+          announcementDetails.to,
+          announcementDetails.value,
+          announcementDetails.data,
+          announcementDetails.operation,
+          announcementDetails.nonce,
+        );
```
Comment (mmv08) [inline src/components/rebirth.tsx:185]: actual change

File: src/components/rescuers.tsx [hunk 3]
```diff
@@ -23,91 +23,101 @@ const numberZoHHMMSS = (timestamp: number) => {
...
+      try {
+        await module.updateConfig(announcer, 0, false);
+      } catch (e) {
         console.error(e);
-    }
-  }, [sdk, module])
+      }
+    },
+    [module],
+  );
```
Comment (mmv08) [inline src/components/rescuers.tsx:57]: actual change

File: src/components/dashboard.tsx [hunk 4]
```diff
@@ -1,73 +1,67 @@
...
       try {
-        const modules = await manager.getModules()
-        console.log(modules)
-        console.log(module.address)
-        const enabled = modules.includes(module.address)
-        setModuleStatus({ label: enabled ? "enabled" : "disabled", showEnableButton: !enabled } )
+        const modules = await manager.getModules();
+        console.log(modules);
+        console.log(module.address);
+        const enabled = modules.includes(module.address);
```
Comment (mikheevm) [inline src/components/dashboard.tsx:30]: actual change

**Review by mmv08 — COMMENTED** [review]
Adding comments to the most important changes as there's a lot of formatting changes

File: src/components/dashboard.tsx [hunk 5]
```diff
@@ -1,73 +1,65 @@
...
-        await sdk.txs.send({
-            txs: [
-                {
-                    to: safe.safeAddress,
-                    value: '0',
-                    data: manager.interface.encodeFunctionData("enableModule", [module.address])
-                },
-            ],
-        });
+      await manager.enableModule(module.address);
```
Comment (mmv08) [inline src/components/dashboard.tsx:42]: actual change

File: src/App.tsx [hunk 6]
```diff
@@ -22,36 +22,41 @@ const Container = styled.form`
...
 const App: React.FC = () => {
   const { sdk, safe } = useSafeAppsSDK();
-  const [selectedTab, setSelectedTab] = useState("dashboard");
+  const [selectedTab, setSelectedTab] = useState('dashboard');
   const module = useMemo(() => {
-    console.log("create module")
-    return new ethers.Contract(DelayedTxModule.address, DelayedTxModule.abi, new SafeAppsSdkProvider(safe, sdk));
-  }, [sdk, safe])
+    console.log('create module');
+    return new ethers.Contract(DelayedTxModule.address, DelayedTxModule.abi, new SafeAppsSdkSigner(safe, sdk));
```
Comment (rmeissner) [inline src/App.tsx]: Couldn't we memoize the signer and use it in both the module and manager instance?

File: src/components/dashboard.tsx [hunk 7]
```diff
@@ -1,73 +1,65 @@
...
-
-  return (<>
-    <Title size="md">Dashboard</Title>
-    <div>Name: {moduleName}</div>
-    <div>Status: {moduleStatus.label} {!moduleStatus.showEnableButton || <Button size="md" color="primary" onClick={enableModule}>Enable</Button>}</div>
-    <Rescuers module={module} enabled={true} />
+  }, [manager, module.address]);
 
+  return (
+    <>
```
Comment (rmeissner) [inline src/components/dashboard.tsx:49]: This is way nicer formatter now :D

**Review by rmeissner — APPROVED** [review]

**Pull request merged**